cartoframes/data/enrichment/enrichment_service.py

```diff
@@ -111,38 +152,77 @@ def __process_filters(filters_dict):
     return filters
 
 
-def __get_tables_and_variables(variables):
+def __process_agg_operators(agg_operators, variables):
+    agg_operators_result = agg_operators.copy()
 
-    if isinstance(variables, pd.Series):
-        variables_id = [variables['id']]
-    elif isinstance(variables, pd.DataFrame):
-        variables_id = variables['id'].tolist()
-    else:
-        raise EnrichmentException('Variable(s) to enrich should be an instance of Series or DataFrame')
+    for variable in variables:
+        variable_name = variable.id.split('.')[-1]
+
+        if variable_name not in agg_operators_result:
+            agg_operators_result[variable_name] = variable.agg_method
+
+    return agg_operators_result
 
-    table_to_variables = __process_enrichment_variables(variables_id)
-    table_data_enrichment = list(table_to_variables.keys()).pop()
-    table_geo_enrichment = __get_name_geotable_from_datatable(table_data_enrichment)
-    variables_list = list(table_to_variables.values()).pop()
 
-    return table_data_enrichment, table_geo_enrichment, variables_list
+def __get_tables_and_variables(variables, user_dataset):
 
+    table_to_geotable, table_to_variables, table_to_project, table_to_dataset =\
+        __process_enrichment_variables(variables, user_dataset)
 
-def __process_enrichment_variables(variables):
+    return table_to_geotable, table_to_variables, table_to_project, table_to_dataset
+
+
+def __process_enrichment_variables(variables_id, user_dataset):
+    table_to_geotable = dict()
     table_to_variables = defaultdict(list)
+    table_to_project = dict()
+    table_to_dataset = dict()
 
-    for variable in variables:
-        variable_split = variable.split('.')
-        table, variable = variable_split[-2], variable_split[-1]
+    for variable_id in variables_id:
+        variable_split = variable_id.id.split('.')
```

Comment:
is `variable_id` a `Variable` instance? If so, I'd change the name to `variable` and move the split to a method in the Variable class.

Reply:
Done!